subprojects/build-init/src/main/groovy/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -63,6 +64,7 @@
     private PathBase distributionBase = PathBase.GRADLE_USER_HOME;
     private String distributionUrl;
     private GradleVersion gradleVersion;
+    private String distType;
```

Comment:
Please change the type of distType to be an enum - as name I would suggest `DistributionType`. Then the logic to check if the distribution type should be simpler. The enum can be an inner class of Wrapper like `PathBase`.